heartbeat/aws-vpc-move-ip

```diff
@@ -213,25 +224,34 @@ ec2ip_validate() {
 }
 
 ec2ip_monitor() {
-        MON_RES=""
+  MON_RES=""
+  if ocf_is_true ${OCF_RESKEY_eni_id_lookup}; then
+    EC2_ATTRIBUTE_LOOKUP="NetworkInterfaceId"
+    EC2_ID="$(ec2ip_get_instance_eni)"
+    identification="interface"
+  else
+    EC2_ATTRIBUTE_LOOKUP="InstanceId"
+    EC2_ID="$EC2_INSTANCE_ID"
```

Comment:
I think it'd be better to do this if/else section where `EC2_INSTANCE_ID` is defined. That way you wont have to do the EC2_INSTANCE_ID to EC2_ID part of the changes.

Remember to update the indentation of this section to use tabs like in the rest of the agent.

Reply:
I get what you are saying, but more variables are set based on this check.
So moving this out of here would loose the other vars like identification.
Maybe even better to drop the whole instance-id part because it's not needed, the eni will always work.

Reply:
If eni also works for older awscli that sounds like a good idea (we dont want to break it for users who upgrade the agent while having older awscli).

Reply:
What is the lowest version of the awscli that you want to support? I can do some testing because I really think we can drop the instance logic. Have to admit that it looks nicer in the logs, but other than that it's overhead..

Reply:
1.14.28 is the oldest we use in RHEL, but maybe some other distro might have a LTS version that provides even older awsli.

Reply:
so what is your suggestion here? shall we leave it in and be safe on the backwards compatibility or test this with the version 1.14.28 of the aws-cli and remove the code if possible. (looking and the aws repo the NetworkInterfaceId is in there, so should work fine I think)

Reply:
Yep, we need to leave it backward compatible until we decide to put a minimal requirement on the AWS CLI, which as of today isn't in place, but it is a good call.

We also need to ensure that whatever feature we're adding, keeps and most importantly, doesn't break anything.